tests/test_core.py

```diff
@@ -386,56 +387,66 @@ def run(self) -> None:
 
 async def test_stage_shutdown(hass: HomeAssistant) -> None:
     """Simulate a shutdown, test calling stuff."""
+    test_stopping = async_capture_events(hass, EVENT_HOMEASSISTANT_STOPPING)
     test_stop = async_capture_events(hass, EVENT_HOMEASSISTANT_STOP)
     test_final_write = async_capture_events(hass, EVENT_HOMEASSISTANT_FINAL_WRITE)
     test_close = async_capture_events(hass, EVENT_HOMEASSISTANT_CLOSE)
     test_all = async_capture_events(hass, MATCH_ALL)
 
     await hass.async_stop()
 
+    assert len(test_stopping) == 1
     assert len(test_stop) == 1
     assert len(test_close) == 1
     assert len(test_final_write) == 1
-    assert len(test_all) == 2
```

Comment:
Try this:

`with patch.object(hass.timeout, "async_timeout", side_effect=asyncio.TimeoutError):
        await hass.async_stop()

    assert hass.state == CoreState.stopped`